panel/widgets/tables.py

```diff
@@ -712,7 +712,10 @@ def stream(self, stream_value, rollover=None, reset_index=True):
             if reset_index:
                 stream_value = stream_value.reset_index(drop=True)
                 stream_value.index += value_index_start
-            combined = pd.concat([self.value, stream_value])
+            if self.value.empty:
+                combined = stream_value
```

Comment:
Could be a problem if the stream_value does not contain all columns. I don't know if this is possible here.